src/test/scala/org/broadinstitute/dsde/firecloud/webservice/PermissionReportApiSpec.scala

```diff
@@ -0,0 +1,260 @@
+package org.broadinstitute.dsde.firecloud.webservice
+
+import org.broadinstitute.dsde.firecloud.FireCloudExceptionWithErrorReport
+import org.broadinstitute.dsde.firecloud.dataaccess.{MockAgoraDAO, MockRawlsDAO}
+import org.broadinstitute.dsde.firecloud.model.ErrorReportExtensions.FCErrorReport
+import org.broadinstitute.dsde.firecloud.model.MethodRepository._
+import org.broadinstitute.dsde.firecloud.model.{MethodConfigurationName, PermissionReport, PermissionReportRequest, UserInfo, WithAccessToken}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.service.{BaseServiceSpec, PermissionReportService, WorkspaceService}
+import org.broadinstitute.dsde.rawls.model.{MethodConfigurationShort, MethodRepoMethod, _}
+import org.scalatest.BeforeAndAfterEach
+import spray.http.{HttpMethods, HttpResponse}
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.Future
+
+class PermissionReportApiSpec extends BaseServiceSpec with WorkspaceApiService with BeforeAndAfterEach {
+
+  def actorRefFactory = system
+
+  val testApp = app.copy(agoraDAO=new PermissionReportMockAgoraDAO(), rawlsDAO=new PermissionReportMockRawlsDAO())
+
+  val workspaceServiceConstructor: (WithAccessToken) => WorkspaceService = WorkspaceService.constructor(testApp)
+  val permissionReportServiceConstructor: (UserInfo) => PermissionReportService = PermissionReportService.constructor(testApp)
+
+  def permissionReportPath(ns:String,name:String) = s"/api/workspaces/$ns/$name/permissionReport"
+
+  "Permission Report API" - {
+
+    "should only accept POST" in {
+      List(HttpMethods.GET, HttpMethods.PUT, HttpMethods.PATCH, HttpMethods.DELETE) map {
+        method =>
+          new RequestBuilder(method)(permissionReportPath("foo","bar")) ~> dummyUserIdHeaders("1234") ~> sealRoute(workspaceRoutes) ~> check {
+            status should equal(MethodNotAllowed)
+          }
+      }
+    }
+
+    "should reject malformed input" in {
+      // endpoint expects an object; send it an array
+      val badPayload = Seq("one","two","three")
```

Comment:
This behavior - ignoring unrecognized keys in a json object when deserializing - is a feature of spray-json and is present in any of our APIs that accepts a json object in its request, unless we explicitly check for this. So from that perspective, it's expected behavior. It's also explicitly why I tested "malformed input" by sending something that wasn't a json object at all.

Try it in /api/library/search or /api/library/suggest, for instance.